### PR TITLE
Support digits in unquoted pg array strings

### PR DESCRIPTION
**Pull request by mfussenegger**
## Summary of the changes / Why this improves CrateDB

Fixes https://github.com/crate/crate/issues/11026

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: libs/pgwire/src/main/antlr/PgArray.g4 [hunk 1]
```diff
@@ -22,40 +22,41 @@ array
...
+
+QSTRING
+    : DIGIT* CHAR+
     ;
 
+CHAR
+    : [a-zA-Z]
+    | [!-+]
+    | '-'
+    | '_'
```
Comment (mfussenegger) [inline libs/pgwire/src/main/antlr/PgArray.g4:52]: Not sure if this is too restrictive, but if much more is allowed thing start to break

Comment (seut) [reply libs/pgwire/src/main/antlr/PgArray.g4:52]: think this is fine at least for now as before it was even more restrictive (e.g. using a `-` as first unquoted char also broke afaik)

File: docs/appendices/release-notes/unreleased.rst [hunk 2]
```diff
@@ -58,6 +58,9 @@ None
 Fixes
 =====
 
+- Fixed an issue in the PostgreSQL wire protocol that would cause
+  de-serialization of arrays to fail if they contained unquoted strings
+  including digits.
```
Comment (seut) [inline docs/appendices/release-notes/unreleased.rst:63]: doesn't this only affect unquoted strings **starting** with a digit?
```suggestion
  starting with digits.
```

**Review by seut — APPROVED** [review]
Lgtm, left a question/suggestion.

File: docs/appendices/release-notes/unreleased.rst (hunk 2 above)
Comment (mfussenegger) [reply docs/appendices/release-notes/unreleased.rst]: Right, updated it :+1:

File: libs/pgwire/src/main/antlr/PgArray.g4 (hunk 1 above)
Comment (mfussenegger) [reply libs/pgwire/src/main/antlr/PgArray.g4:52]: Looks like this caused a regression, previously `.` was allowed, now it isn't anymore

**Pull request merged**